### PR TITLE
[SON-150] list_active_sons wallet_api call 

### PR DESCRIPTION
**Pull request by gladcow**
Adds `list_active_sons` call to wallet API and the unit test for it.

### CODE REVIEW

File: libraries/wallet/wallet.cpp [hunk 1]
```diff
@@ -1932,6 +1932,32 @@ class wallet_api_impl
       return sign_transaction( tx, broadcast );
    } FC_CAPTURE_AND_RETHROW( (owner_account)(broadcast) ) }
 
+   map<string, son_id_type> list_active_sons()
```
Comment (oxarbitrage) [inline libraries/wallet/wallet.cpp]: please change indentation of the whole function to 3 spaces. thanks!

File: libraries/wallet/wallet.cpp [hunk 2]
```diff
@@ -1932,6 +1932,32 @@ class wallet_api_impl
...
+   { try {
+           global_property_object gpo = get_global_properties();
+           std::vector<fc::optional<son_object>> son_objects =
+                   _remote_db->get_sons(gpo.active_sons);
```
Comment (oxarbitrage) [inline libraries/wallet/wallet.cpp]: I think code of this 2 lines is short enough to make it in 1. 120 characters can be a limit that will make the code in github to display without horizontal scrolling in major browsers and screen sizes. We don't have any number defined but in this case, if you want, can consider making it in 1 line.

File: libraries/wallet/wallet.cpp [hunk 3]
```diff
@@ -1932,6 +1932,32 @@ class wallet_api_impl
...
+           vector<account_id_type> owners;
+           owners.resize(son_objects.size());
+           std::transform(son_objects.begin(), son_objects.end(), owners.begin(),
+                          [](const fc::optional<son_object>& obj) {
+                                if(!obj)
+                                    FC_THROW("Invalid active SONs list in global properties.");
```
Comment (oxarbitrage) [inline libraries/wallet/wallet.cpp]: What needs to happen in order to get this exception and the one in line 1955? Should we have a test case for this ?

Comment (gladcow) [reply libraries/wallet/wallet.cpp]: I don't know if it is possible really,  you are right, `FC_ASSERT` is better case here.

**Comment by gladcow** [comment]
@oxarbitrage ,  I have addressed the issues,  can you re-review,  please?

**Pull request merged**